### PR TITLE
Chem colors

### PR DESCRIPTION
**Pull request by aspiringLich**
## About the PR

Changed the colors of many boring gray chemicals. Mostly based off of [CPK Coloring](https://en.wikipedia.org/wiki/CPK_coloring)

<table>
<thead><tr><th>Chemical</th><th>Before</th><th>After</th></tr></thead>
<tr><th>Hydrogen</th><th>

`#808080`

</th><th>

`#cccccc`

</th></tr>
<tr><th>Fluorine</th><th>

`#808080`

</th><th>

`#ba85cc`

</th></tr>
<tr><th>Oxygen</th><th>

`#808080`

</th><th>

`#c4f5ff`

</th></tr>
<tr><th>Nitrogen</th><th>

`#808080`

</th><th>

`#a1e1ff`

</th></tr>
<tr><th>NitrousOxide</th><th>

`#808080`

</th><th>

`#5d88c9`

</th></tr>
<tr><th>Phosphorus</th><th>

`#ede4e4`

</th><th>

`#f5da9a`

</th></tr>
<tr><th>Lithium</th><th>

`#c6c8cc`

</th><th>

`#d6c0a7`

</th></tr>
<tr><th>Sodium</th><th>

`#c6c8cc`

</th><th>

`#a7b3d6`

</th></tr>
</table>

## Why / Balance

Pretty colors :)

Makes it slightly easier to distinguish colors in a chem dispenser.

Most of these chemicals you will rarely see, and even if you do, won't care about, in a spill. The colors are purely cosmetic for the dispenser (/ chem master coming soon i swear ooooooo), so this will not affect gameplay. I left potassium unchanged because janitors actively look out for it, so if it was no longer white that may throw them off.

## Technical details

YAML or something.

## Media

image

## Breaking changes

**Changelog**

:cl:
- tweak: Some chemicals have been made more colorful and child-friendly.

### CODE REVIEW

**Comment by UbaserB** [comment]
chlorine and fluorine are too similar now

**Comment by aspiringLich** [comment]
That could be easily changed, but why is that something to worry about? Before the changes there were like four chemicals with the same medium gray color.

**Comment by UbaserB** [comment]
> That could be easily changed, but why is that something to worry about? Before the changes there were like four chemicals with the same medium gray color.

it's iconic, and doesn't contrast well for CLF3 especially; where the chemical reaction only requires two colours

**Comment by 0x6273** [comment]
This makes no sense why they would use arbitrary colors meant for molecular model instead of their actual real life colors. This doesn't just affect chem ui but puddles as well. Oxygen should be light blue and nitrogen should be clear, for example. Nitrogen having the color of oxygen would be very confusing.

**Comment by aspiringLich** [comment]
Okay, I can see Oxygen being an issue (and I was iffy about that when making the PR), but for literally any chemical you dont see in real life, the colors picked are completely arbitrarily or based on what feels right.

I can see your concerns around fluorine too. Would changing fluorine to a light purple/pink and oxygen to light blue work? I would like to keep Nitrogen as light blue because I just kinda hate using '#808080' to represent colorlessness.

**Comment by Tayrtahn** [comment]
Wondering if it would be worthwhile to add separate `color` and `uiColor` properties.

**Comment by beck-thompson** [comment]
Yeah so many chems being the same color was an issue! I always get confused if two are on the floor.

**Comment by EmoGarbage404** [comment]
yeah i'd go with tayrtahn's suggestion for this one. Include a secondary color that can be used to override the color in the UI and just use the basic one if none is specified. The color for chems is very obviously implied to be the physical color so a lot of this doesn't make much sense.

**Comment by aspiringLich** [comment]
I mean, I think that giving chems somewhat distinct colors can't hurt, just so not absolutely everything is `#808080`. For example, ClF3 is light pink even though really it should be colorless like 99% of other chemicals. And radium is green because, you know, radioactive, when it should be silvery white like every other metal. Why would nitrogen and hydrogen have to stick to their real life colors of, uh, no color? 

Having a different color in the UI of the dispenser also brings other complications, because the color of the jugs is based on the entire solution of the jug, which could be any mix of chems. Having a special-case color for certain chemicals is already kind of unintuitive, but what, you would have to mix the `uiColor`'s of all the chemicals in the solution? It's certainly possible but I don't think it's as clean of a solution as you guys were thinking. It's just way easier and simpler to just represent the color of the solution.

Also my main goal with these changes was to just have chems be distinguishable, having half the jugs be a shade of medium-gray frankly just isn't cool. Making the colors in the chem dispenser be distinguishable only solves half the problem IMO.

**Comment by aspiringLich** [comment]
I'm all for realism, I just don't see why we should follow realism when realism tells us that 90% of the chemicals should be colorless or white. And if we keep it "realistic" for oxygen, we would be making exceptions for chemicals like radium which I hope we can agree makes sense to keep green.

**Comment by 0x6273** [comment]
Maybe jugs could be colorable, and the ui would use that color instead of the chem color? This would also help distinguish jugs outside of the chem dispenser.

File: Resources/Prototypes/Reagents/gases.yml [hunk 1]
```diff
@@ -190,7 +190,7 @@
   desc: reagent-desc-nitrogen
   physicalDesc: reagent-physical-desc-gaseous
   flavor: bitter
-  color: "#808080"
+  color: "#a1e1ff"
```
Comment (0x6273) [inline Resources/Prototypes/Reagents/gases.yml:193]: I think red like the nitrogen tanks/canisters would make more sense than blue?

Comment (aspiringLich) [reply Resources/Prototypes/Reagents/gases.yml:193]: Yeah I thought of that but I just felt like red was a weird color to give to _nitrogen_. Something that makes up most of air not being blue just feels wrong.

**Comment by mirrorcult** [comment]
these are fine changes, i dont agree with standardizing in the other direction, theres not really any benefit to making chems realistic colors

**Comment by aspiringLich** [comment]
I honestly didn't expect this PR to be at all controversial; I thought it was pretty clear that a bunch of chems being colored '#808080' was an oversight or placeholder.

**Review by mirrorcult — APPROVED** [review]

**Pull request merged**